frontend/src/metabase-lib/lib/Dimension.js

```diff
@@ -845,22 +845,29 @@ export class AggregationDimension extends Dimension {
   }
 
   column(extra = {}) {
-    const aggregation = this.aggregation();
-    const { special_type, ...column } = super.column();
     return {
-      ...column,
-      // don't pass through `special_type` when aggregating these types
-      ...(!UNAGGREGATED_SPECIAL_TYPES.has(special_type) && { special_type }),
-      base_type: aggregation ? aggregation.baseType() : TYPE.Float,
+      ...super.column(),
       source: "aggregation",
       ...extra,
     };
   }
 
   field() {
-    // FIXME: it isn't really correct to return the unaggregated field. return a fake Field object?
-    const dimension = this.aggregation().dimension();
-    return dimension ? dimension.field() : super.field();
+    const aggregation = this.aggregation();
+    if (!aggregation) {
+      return super.field();
+    }
+    const dimension = aggregation.dimension();
+    const field = dimension && dimension.field();
+    const { special_type } = field || {};
+    return new Field({
+      display_name: aggregation.displayName(),
```

Comment:
I think this is missing things like `name`.

Reply:
I just added name. What else would you put in here? Maybe `field_ref` referencing the name?

Reply:
@tlrobinson do you think adding name is enough here? I'm not sure where these properties are being used.

Reply:
Honestly I'm not sure. Grepping for `\.field\(\)` it looks like we primarily use it to check for specific types. Most other use-cases should probably be replaced with dimension methods directly.

The other place I thought it was used by `StructuredQuery`'s `table()` for nested questions but it looks like that does `new Field({ ...column, ... })` (and `column` gets `name` from `dimension.columnName()`)

I think the main thing I want to make sure is correct is `column()`, and the only thing it uses from field are `id`, `base_type`, and `special_type`. Ultimately we should have integration tests comparing the actual `column` with the result of `dimension.column()` for various queries. This is fine for now though.